src/main/java/net/dries007/tfc/objects/fluids/FluidsTFC.java

```diff
@@ -219,12 +219,18 @@ public static void registerFluids()
     @Nonnull
     private static FluidWrapper registerFluid(@Nonnull Fluid newFluid)
     {
-        boolean isDefault = FluidRegistry.registerFluid(newFluid);
+        boolean isDefault = !FluidRegistry.isFluidRegistered(newFluid.getName());
+
         if (!isDefault)
         {
             // Fluid was already registered with this name, default to that fluid
             newFluid = FluidRegistry.getFluid(newFluid.getName());
         }
+        else
+        {
+            // No fluid found we are safe to register our default
+            FluidRegistry.registerFluid(newFluid);
+        }
```

Comment:
I'm intrigued. This seems like it may well be the cause of #840 ?

Reply:
Sorry messed up the PR